sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleController.java

```diff
@@ -241,8 +241,16 @@ private void requestSampleCollectionTaskCreation(SampleDto dto, SampleEditForm f
 			@Override
 			public void buttonClick(ClickEvent event) {
 				popupWindow.close();
-				ControllerProvider.getTaskController().createSampleCollectionTask(TaskContext.CASE,
-						dto.getAssociatedCase() != null ? dto.getAssociatedCase() : dto.getAssociatedContact(), dto);
+				final CaseReferenceDto associatedCase = dto.getAssociatedCase();
+				if (associatedCase != null) {
+					ControllerProvider.getTaskController().createSampleCollectionTask(TaskContext.CASE,
+							associatedCase, dto);
+				}
+				final ContactReferenceDto associatedContact = dto.getAssociatedContact();
+				if (associatedContact != null) {
+					ControllerProvider.getTaskController().createSampleCollectionTask(TaskContext.CONTACT,
+							associatedContact, dto);
+				}
```

Comment:
I don't think two tasks should be generated because they're basically redundant. If a sample is assigned to both a case and a contact, only create a sample collection task for the case association.